src/aws-cpp-sdk-core/source/utils/crypto/PrecalculatedHash.cpp

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+
+#include <aws/core/utils/crypto/PrecalculatedHash.h>
+
+using namespace Aws::Utils::Crypto;
+
+PrecalculatedHash::PrecalculatedHash(const Aws::String &mHash): m_hash(ByteBuffer((unsigned char*)mHash.c_str(),mHash.length())){
+m_hash_string = mHash;
```

Comment:
add to the initializer list instead of assigning in the ctor and rename to m_hashString pls